ddtrace/debugging/_probe/model.py

```diff
@@ -227,73 +162,30 @@ def eval(self, _locals):
 
 @attr.s
 class ExpressionTemplateSegment(TemplateSegment):
-    expr = attr.ib(type=DslExpression, default=None)  # type: DslExpression
+    expr = attr.ib(type=DDExpression, default=None)  # type: DDExpression
 
     def eval(self, _locals):
         # type: (Dict[str,Any]) -> Any
         return self.expr.eval(_locals)
 
 
 @attr.s
-class SnapshotProbeDetails(six.with_metaclass(abc.ABCMeta)):
-    capture = attr.ib(type=CaptureLimits, eq=False)  # type: CaptureLimits
-
-
-def snapshot_probe_defaults(f):
-    def _wrapper(*args, **kwargs):
-        kwargs.setdefault("capture", CaptureLimits())
-        return f(*args, **kwargs)
-
-    return _wrapper
+class LogProbeMixin(object):
+    template = attr.ib(type=str)
+    segments = attr.ib(type=List[TemplateSegment])
+    capture_snapshot = attr.ib(type=bool)
```

Comment:
`capture_snapshot` feels a bit like two repeated words. How about `has_snapshot`, or `take_snapshot`, or simply `snapshot`/`capture`?

Reply:
renamed to take_snapshot